components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -403,36 +406,74 @@ class BrowserToolbar @JvmOverloads constructor(
      * @param listener Callback that will be invoked whenever the checked state changes.
      */
     open class ToggleButton(
-        @DrawableRes imageResource: Int,
-        @DrawableRes imageResourceSelected: Int,
+        image: Drawable,
+        imageSelected: Drawable,
         contentDescription: String,
         contentDescriptionSelected: String,
         visible: () -> Boolean = { true },
         selected: Boolean = false,
-        @DrawableRes background: Int? = null,
-        private val padding: Padding = defaultActionPadding,
+        @DrawableRes background: Int = 0,
+        val padding: Padding = DEFAULT_PADDING,
         listener: (Boolean) -> Unit
     ) : Toolbar.ActionToggleButton(
-        imageResource,
-        imageResourceSelected,
+        image,
+        imageSelected,
         contentDescription,
         contentDescriptionSelected,
         visible,
         selected,
         background,
-        listener = listener
+        padding,
+        listener
     ) {
+
         override fun createView(parent: ViewGroup): View {
             return super.createView(parent).apply {
                 setPadding(padding)
             }
         }
     }
 
+    /**
+     * An action that either shows an active button or an inactive button based on the provided
+     * <code>isLoading</code> lambda.
+     */
+    open class TwoStateButton(
+        private val enabledImage: Drawable,
+        private val enabledContentDescription: String,
+        private val disabledImage: Drawable,
+        private val disabledContentDescription: String,
+        private val isLoading: () -> Boolean = { true },
```

Comment:
We'd need to find a different name for this lambda too! :)

Reply:
Oops, fixed!